### PR TITLE
feat: [PIPE-16006]: Add information regarding execution context

### PR DESCRIPTION
**Pull request by krishi0408**
Thanks for contributing to the Harness Developer Hub! Our code owners will review your submission.

## Description

* Please describe your changes: \__________________________________
* Jira/GitHub Issue numbers (if any): \______________________________
* Preview links/images (Internal contributors only): \__________________

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [ ] Successful preview build.
- [ ] Code owner review.
- [ ] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

**Review by SudarshanPurohit — APPROVED** [review]
Overall looks good, just wanted to suggest two more points: 

* The execution context is not editable. Users cannot add or remove items from it.
* The purpose of the execution context is to help debug previous executions of the pipeline. This is an additional tool on top of the Input/Output variables that are listed for each of the steps.

File: docs/platform/variables-and-expressions/harness-variables.md [hunk 1]
```diff
@@ -432,6 +432,58 @@ If the variable is local to a scope within the pipeline, such as a stage or step
 
 ![](./static/harness-variables-16.png)
 
+### Get input/output expressions used in a from execution context
+
+You can get the information about the resolved expressions along with it's value for each step in Step detail pane.
```
Comment (roshnisarangadharan) [inline docs/platform/variables-and-expressions/harness-variables.md:437]:
```suggestion
You can get the information about the resolved expressions along with its value for each step in the **Step Detail** pane.
```

File: docs/platform/variables-and-expressions/harness-variables.md [hunk 2]
```diff
@@ -432,6 +432,58 @@ If the variable is local to a scope within the pipeline, such as a stage or step
...
+
+Let's consider a following example:
```
Comment (roshnisarangadharan) [inline docs/platform/variables-and-expressions/harness-variables.md:439]:
```suggestion
Let's consider the following example:
```

File: docs/platform/variables-and-expressions/harness-variables.md [hunk 3]
```diff
@@ -432,6 +432,58 @@ If the variable is local to a scope within the pipeline, such as a stage or step
...
+                      - name: executionUrl
+                        type: String
+                        value: <+pipeline.executionUrl>
+                    outputVariables: []
+                  timeout: 10m
+        tags: {}
+```
+![](./static/executionContext.png)
+
+**Execution context** is a table of keys and values, with the keys being the expressions that were referred to within the step. In the above example, in step **ShellScript_1** we have defined an Input/environment variable `executionUrl` with an expression type. Once you run the pipeline, click on the step **ShellScript_1** and select Execution context in the step details pane.
```
Comment (roshnisarangadharan) [inline docs/platform/variables-and-expressions/harness-variables.md:477]:
```suggestion
**Execution context** is a table of keys and values with keys being the expressions that were referred within the step. In the above example, in the step, **ShellScript_1**, we have defined an input/environment variable, `executionUrl` with an expression type. Once you run the pipeline, click on the **ShellScript_1** step and then select Execution Context in the Step Details pane.
```

File: docs/platform/variables-and-expressions/harness-variables.md [hunk 4]
```diff
@@ -432,6 +432,58 @@ If the variable is local to a scope within the pipeline, such as a stage or step
...
+
+You can also get expressions for **step level variable** and execution inputs from the **Execution Context**.
```
Comment (roshnisarangadharan) [inline docs/platform/variables-and-expressions/harness-variables.md:479]:
```suggestion
You can also get the expressions for **step level variable** and execution inputs from **Execution Context**.
```

File: docs/platform/variables-and-expressions/harness-variables.md [hunk 5]
```diff
@@ -432,6 +432,58 @@ If the variable is local to a scope within the pipeline, such as a stage or step
...
+
+![](./static/expression_executionContext.png)
+
+:::info note
+Please note that secrets will not be displayed in the Execution context.
```
Comment (roshnisarangadharan) [inline docs/platform/variables-and-expressions/harness-variables.md:484]:
```suggestion
Secrets are not displayed in Execution Context.
```

**Review by roshnisarangadharan — DISMISSED** [review]

**Comment by krishi0408** [comment]
> Overall looks good, just wanted to suggest two more points:
> 
> * The execution context is not editable. Users cannot add or remove items from it.
> * The purpose of the execution context is to help debug previous executions of the pipeline. This is an additional tool on top of the Input/Output variables that are listed for each of the steps.

sure @SudarshanPurohit added these points as well.

**Pull request merged**